soql-standalone-parser/src/main/scala/com/socrata/soql/ast/Expression.scala

```diff
@@ -399,8 +349,18 @@ case class WindowFunctionInfo(partitions: Seq[Expression], orderings: Seq[OrderB
 
 case class Hole(name: HoleName)(val position: Position) extends Expression {
   def allColumnRefs = Set.empty
-  protected override def asString = "?" + name
-  def format(depth: Int, sb: StringBuilder, limit: Option[Int]) = Some(sb.append(asString))
 
   def replaceHoles(f: Hole => Expression) = f(this)
+
+  def doc = Doc("?" + name)
+}
+
+object bleh extends App {
```

Comment:
Was this for testing?

Reply:
Yeah, oops.  Well, kinda.  This is the App that powers that demo site I put up to show the formatter off. :)